Datadog/Example/Scenarios/Tracing/URLSessionAutoInstrumentation/TracingURLSessionViewController.swift

```diff
@@ -0,0 +1,63 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import UIKit
+
+internal class TracingURLSessionViewController: UIViewController {
+    private var testScenario: TracingURLSessionScenario!
+    private let session = URLSession.shared
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        self.testScenario = (appConfiguration.testScenario as! TracingURLSessionScenario)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        callSuccessfullURL {
+            self.callSuccessfullURLRequest {
+                self.callBadURL {
+                    self.useNotInstrumentedAPIs()
+                }
+            }
+        }
+    }
+
+    private func callSuccessfullURL(completion: @escaping () -> Void) {
+        let task = session.dataTask(with: testScenario.customGETResourceURL) { _, _, error in
+            assert(error == nil)
+            completion()
+        }
+        task.resume()
+    }
+
+    private func callSuccessfullURLRequest(completion: @escaping () -> Void) {
+        let task = session.dataTask(with: testScenario.customPOSTRequest) { _, _, error in
+            assert(error == nil)
+            completion()
+        }
+        task.resume()
+    }
+
+    private func callBadURL(completion: @escaping () -> Void) {
+        let task = session.dataTask(with: testScenario.badResourceURL) { _, _, error in
+            assert(error != nil)
```

Comment:
it's weird that you are doing the assertions in the ViewController directly.

Reply:
It's just sanity. If it's not met in integration tests, this will crash the host app (and fail the test). I added it here, as we use exactly the same setup for debugging our SDK with running the `Example` app directly - in such case I want to have an instant feedback on the assumption (if it has or doesn't have the error).